app/Models/User.php

```diff
@@ -26,4 +26,9 @@ class User extends Authenticatable
     protected $hidden = [
         'password', 'remember_token',
     ];
+
+    public function match() {
+
```

Comment:
this is a many relationship it needs to be called matches. also you need to make it into a belongsToMany - not has many - and you need a pivot table - https://laravel.com/docs/5.4/eloquent-relationships#many-to-many (if you want we can discuss this)